Core/CoreTests/Features/Files/View/FileList/FileListViewControllerTests.swift

```diff
@@ -271,6 +278,51 @@ class FileListViewControllerTests: CoreTestCase {
         XCTAssertEqual(controller.tableView.numberOfRows(inSection: 0), 0)
     }
 
+    func test_addFileButtonVisible_whenNotRestricted() {
+        studentAccessInteractor.setRestricted(false)
+
+        controller = FileListViewController.create(
+            env: environment,
+            context: .currentUser,
+            path: "Folder A",
+            studentAccessInteractor: studentAccessInteractor
+        )
+
+        controller.view.layoutIfNeeded()
+        controller.viewWillAppear(false)
+
+        _ = controller.addButton.target?.perform(controller.addButton.action)
+        let sheet = router.presented as? BottomSheetPickerViewController
+        let titles = sheet?.actions.map { $0.title } ?? []
+
+        XCTAssertTrue(titles.contains("Add File"), "Add File should be visible when not restricted")
+        XCTAssertTrue(titles.contains("Add Folder"), "Add Folder should always be visible")
+    }
+
+    func test_addFileButtonHidden_whenRestricted() {
+        studentAccessInteractor.setRestricted(true)
+
+        controller = FileListViewController.create(
+            env: environment,
+            context: .currentUser,
+            path: "Folder A",
+            studentAccessInteractor: studentAccessInteractor,
+            scheduler: testScheduler.eraseToAnyScheduler()
+        )
+
+        controller.view.layoutIfNeeded()
+        controller.viewWillAppear(false)
+
+        // Move virtual time forward by 100ms
+        testScheduler.advance(by: .seconds(0.1))
+
```

Comment:
Thanks, this way the test is more deterministic. Since we are not really interested in how much time have elapsed, we can use az immediate scheduler to further simplify the test and remove this advance call.

```
    private var testScheduler: AnySchedulerOf<DispatchQueue>!

    override func setUp() {
        super.setUp()
        testScheduler = DispatchQueue.immediate.eraseToAnyScheduler()
```
```suggestion
```